[GH-3947]: Added confirmation dialog when deleting a card in calendar/galley view

varunKT001 opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute

Assign two reviewers for this pull request from the names suggested. If no names are suggested or you are not sure who to assign, set `Core Focalboard` as the reviewer.
-->

#### Summary

Added confirmation dialog when deleting a card in calendar/galley view.

<!--
A description of what this pull request does, as well as QA test steps (if applicable and if not already added to the ticket).
-->

#### Ticket Link

Fixes #3947 

<!--
If this pull request addresses a Github issue, please link the relevant issue, e.g.

  Fixes https://github.com/mattermost/focalboard/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

mattermod commented:
Hello @varunKT001,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

Pinjasaur commented:
@varunKT001 thanks for your contribution! PR looks to in a good spot, just appears to need a bit of tweaking to pass CI.

To simulate the webapp CI runs locally you can use `make webapp-ci`. It's looking like you'll need to update a Jest snapshot, which can be done by running `npm run updatesnapshot` from the `webapp` directory. Finally, since you made some code changes that reference i18n translation strings, can you run `npm run i18n-extract` also from the `webapp` directory. Thanks!

Rajat-Dabade commented:
@varunKT001, the changes look good to me as well. I do agree with @Pinjasaur suggestion and once you are done with the suggested modification, which I guess will fix all the CI issues, we can merge it. 

Thanks for the contribution. Nice work 👍

varunKT001 commented:
@Pinjasaur @Rajat-Dabade

I have updated the Jest snapshot by running `npm run updatesnapshot`. I have also run the command `npm run i18n-extract` to update the i18n translation strings. Please review and suggest if there are any changes needed.

Thanks for helping 😄

Rajat-Dabade commented:
@varunKT001, thanks for implementing the suggested changes.

It seems to be we have CI failing again, I suspect it is due to the additional confirmation Dialog box for deleting the card (changes in this PR). You can just update the test case and try running `npm run test` in `webapp` locally if the tests are passing or not.

varunKT001 commented:
@Rajat-Dabade 

I have updated the test cases and run `npm run test`. There are no errors in my local setup. I have also modified the code a little bit. Please review and suggest if any changes are needed.

Thank you.

varunKT001 commented:
@Pinjasaur @Rajat-Dabade 

I have updated the test cases and run `npm run test`. There are no errors in my local setup. 

![image](https://user-images.githubusercontent.com/83509023/195344920-5efecbef-f7a8-4338-b570-ef5494515b74.png)

I have also modified the code a little bit. Please review and suggest if any changes are needed.

Thank you.

Rajat-Dabade commented:
@varunKT001, I think we have a different issue now to be solved first, the CI is failing due to some linter issue in `fullCalendar.tsx`. Please run `make webapp-ci` in `focalboard` directory. 

Also, the overlay for the Dialog in the `calendar` view for confirming deletion is not covering the entire screen, which it should. 

![Screenshot 2022-10-12 at 8 03 12 PM](https://user-images.githubusercontent.com/26198062/195371375-3704c8de-81db-4ea7-9585-ad951d93dc38.png)

varunKT001 commented:
@Rajat-Dabade @Pinjasaur 

I have fixed the linter errors as well as the overlay. I have run the `make webapp-ci` in my local setup and GitHub actions on my forked repository, and it is succeeding. Please review and suggest if any changes are needed.

![image](https://user-images.githubusercontent.com/83509023/195384750-7f2520b9-9302-4adb-9dea-4a0e8e1f1650.png)

Thank you.

varunKT001 commented on webapp/src/components/markdownEditor.test.tsx:
```diff
@@ -14,6 +14,7 @@ import {MarkdownEditor} from './markdownEditor'
 jest.mock('../utils')
 jest.useFakeTimers()
 jest.mock('draft-js/lib/generateRandomKey', () => () => '123')
+jest.setTimeout(30000)
```
A test in `markdownEditor.test.tsx` was taking too long to complete and thus generated error. That's why I put a a timeout to increase the default timeout of 5000ms.

Rajat-Dabade replied:
It's strange, it worked for me in my local. Anyways it doesn't matter that much. Let's stick to the default timeout offered by jest to avoid unnecessary changes.

varunKT001 commented on webapp/i18n/en.json:
```diff
@@ -1,423 +1,2 @@
 {
-  "AppBar.Tooltip": "Toggle Linked Boards",
```
I ran the command before committing the changes. Don't know much about `i18n` but these were the changes made when I ran the command.

varunKT001 replied:
@Rajat-Dabade Should I add it manually?

Pinjasaur replied:
@varunKT001 I tested locally and am not seeing all the data from `en.json` gone. If needed, we can manually run `i18n-extract` for you on this PR but I'd like to run through a few steps before we go that far:

1. confirm you're using the correct Node/npm by running `nvm install` from the `webapp` directory
2. if needed, re-install dependencies (the `i18n-extract` is using `formatjs` under the hood, which is installed as a dep)
3. `cd` to the `webapp` directory and run `npm run i18n-extract`

If you're still having trouble I'd be great if you could share a bit more about any error messages you might be receiving and what environment you're using to contribute on. It might simply be a gap in our documentation. Let us know, thanks!

varunKT001 replied:
@Pinjasaur @Rajat-Dabade 

I re-installed all the dependencies and run the `npm run i18n-extract` but there is still no data in the `en.json`.

Node version: 16.13.0
NPM version: 8.1.3
Operating system: Windows 10 (21H1)

![image](https://user-images.githubusercontent.com/83509023/195505810-7a6c0878-7d40-47ef-a138-63476ef8530a.png)

There are no errors or warnings thrown when I run the command, so I can't figure out what's the issue here 😕

varunKT001 replied:
@Pinjasaur @Rajat-Dabade For now, should I manually add the content to the `en.json` file?

Pinjasaur replied:
@varunKT001 I did an `i18n-extract` locally and pushed it here for your PR. Apologies that the command isn't working for you. I'm guessing it's a Windows issue, even though based on your screenshot that looks like Git Bash which I would normally expect to work. It's bizarre there's no error. I wonder if the paths aren't being interpreted right due to filesystem differences. I'll file an issue to track it and see if we can get to the bottom of it.

Pinjasaur replied:
While writing up #4017 I found #3932 which might fix the problems on Windows. Need to investigate further.

Rajat-Dabade replied:
Okay, got it. Thanks, @Pinjasaur.

varunKT001 replied:
@Pinjasaur 

I don't know why it's working unexpectedly. It happens very rarely that I face problems like this due to Windows. I'm thinking of using WSL for development purposes. 

BTW thanks for helping 😄

varunKT001 commented:
@Pinjasaur @Rajat-Dabade 

I re-installed all the dependencies and run the `npm run i18n-extract` but there is still no data in the `en.json`.

Node version: 16.13.0
NPM version: 8.1.3

![image](https://user-images.githubusercontent.com/83509023/195505810-7a6c0878-7d40-47ef-a138-63476ef8530a.png)

There are no errors or warnings thrown when I run the command, so I can't figure out what's the issue here 😕

Rajat-Dabade reviewed (approved):
LGTM, nice work @varunKT001 🚀. Thanks for the contribution, really appreciate your work.

varunKT001 commented:
@Pinjasaur @Rajat-Dabade 

Thank you for helping me throughout the process. I learned many new things, like the proper use of hooks like `useCallback`, `useMemo`, etc., and especially the importance of testing. It was my first contribution to such a big project.

Looking forward to contributing to more such issues 🚀
